fdbserver/Coordination.actor.cpp

```diff
@@ -228,7 +228,9 @@ ACTOR Future<Void> openDatabase(ClientData* db,
 				if (req.supportedVersions.size() > 0) {
 					db->clientStatusInfoMap.erase(req.reply.getEndpoint().getPrimaryAddress());
 				}
-				replyContents = db->clientInfo->get();
+				if (db->clientInfo->get().read().id.isValid()) {
+					replyContents = db->clientInfo->get();
+				}
 				break;
```

Comment:
Should this break inside the above `if`?

Reply:
No, this is in a wait(timeout), so we should break anyways. When the id is invalid, the `replyContents` is `default_error_or()`.

Reply:
Got it. Can you add a comment that this will cause an error to be returned?

Reply:
I got confused myself now. @sfc-gh-etschannen should we move the `break` into the above `if`? In other words, what is this `SERVER_KNOBS->CLIENT_REGISTER_INTERVAL` for? When we have waited `SERVER_KNOBS->CLIENT_REGISTER_INTERVAL`, do we want to reply even if `db->clientInfo->get().read().id` is not valid (in this case, replyContents will be `default_error_or`). If we use `SERVER_KNOBS->CLIENT_REGISTER_INTERVAL` as a timeout, then what's the difference between it and the above `wait(checkStuck)`?